### DISCUSSION TITLE
Update on Status, Discussion on the Process?

### DISCUSSION THREAD

**Discussion by kirkw** [orioledb/orioledb#238] [General]
First, Kudos for embarking on this.
One of the biggest losses for us was IOT tables.  We use them for logging.  They go back forever.
While we plan to partition them in PG, the horrible part is the extra space of indexing them.  We
rely on a certain level of uniqueness...

So, our largest tables now take double the storage in regular PG.

The upside is that we have "large" datasets I can load into OrioleDB type tables...
What can I do, with regards to that, that can help you?  (We have about 4 of these tables.  About 1 bln or more rows each)
and we are looking now to partition them.

What steps can I do to help provide you feedback/information that you need?

Realizing that this might not be there until PG 20...  But better late than never.
I am also curious to see how Orioledb works w/regrads to partitioned tables.

Finally, your originally stated goals are "feeling dated".  It would be nice to see where we are at
inside of the bigger timeline... (Past proof of concept into debugging/validating, or are we already in Beta?)

**Comment by akorotkov** [comment]
Hi, Kirk!

I appreciate your interest in OrioleDB.
We are approaching beta. It should come out on June 1st.
OrioleDB should be very good for your use case.  Indeed, index-organized tables can significantly reduce the disk space occupied.  I can also advise you to try [table compression](https://github.com/orioledb/orioledb/blob/main/doc/usage.md#block-level-data-compression) to reduce the data footprint further.  You're very welcome to try your workload either on alpha13 or beta1 (which will come out very soon) and share your feedback.

Regarding synchronization with PostgreSQL Core, our current goal is to become a pure extension of PostgreSQL without Core patch.  We're [making progress on this](https://supabase.com/blog/postgres-pluggable-strorage) reducing the patchset from 5k lines to 2k lines of code.  And I think this goal realistically can be achieved much faster than PG 20.

**Reply by kirkw** [reply]
Hmmm... Reading in more details, I am curious the impact of losing bitmap scans...
We have multi-part key fields.  But worse, the first item in the tables are:
TIMESTAMP, DATE, BIGINT, VARCHAR(9), VARCHAR(20), VARCHAR(30), VARCHAR(200)

With a range of having 2 columns - 9 columns.
And leveraging the UNIQUE key violation on insert to protect against duplicates.

I will wait for beta1.  I think I can test out some of the benefits of compression.
I assume my normal "COPY " file we use to load today will work fine (we mimic pg_dump to produce our import files).

Once we get there, I will DEFINITELY provide feedback.  I will look for guidance so I can "pgbench" or do whatever else to get you data you need.  Although it's a "boatload" of data, I can also load the data in the default PG style.  I have really fast SSDs, so we could actually leverage testing various operations, and then "comparing" to base to make sure we didn't lose anything.

my preference would be to have some scripts that exercise things (turn on compression, test a bunch of access, and then close up with comparing all the data to the original).  Tracking time to compare with original to time to compare with compression.  Maybe even test against OrioleDB table that is not compressed (I like the thought of looping back  to the original table for obvious reasons).

Hopefully you get the changes you need to core made there when you minimize the impact (and maximize the flexibility).

**Comment by FluffyDiscord** [comment]
After so many years of watching, I have to ask, is it still not production ready?

**Reply by kirkcarsley** [reply]
I believe they still need changes to core, in order for this to be "just an extension".  I also believe something changed in where their support was coming from...

**Reply by ImreSamu** [reply]
Based on the latest official information from 4 June 2026: 
*  **"OrioleDB (targeting production-readiness this year)"**

context: 
_"Multigres will be available on the platform when it moves out of alpha. Together with our work **on OrioleDB (targeting production-readiness this year),** we believe that we will have a complete Postgres solution that solves many hairy Postgres problems, including bloat, connection pooling, and Jepsen-level high availability. Sharding is also on the menu - we'll share more details later."_  ( via https://supabase.com/blog/supabase-series-f  04 Jun 2026 ) 

There is also a related new open position:
 - **OrioleDB Developer (AMER)**  https://jobs.ashbyhq.com/supabase/621516b5-4d8f-4ee9-8287-2da12e63d023

**Reply by FluffyDiscord** [reply]
i was looking at the multigres, but i didnt read the whole article, so its tied to that, this will, actually, be absolutely huge
